v4.0.0-alpha3 (#101)
strangelove-ventures/noble-networks: testnet/grand-1/chain-upgrades/v4.0.0-alpha3/README.md

# Upgrade to v4.0.0-alpha3

## Overview

We are excited to announce the next software upgrade for the Noble `grand-1` testnet chain. This upgrade will transition the network to `v4.0.0-alpha3`. This upgrade adds TBD.

**Key Details:**
- **Target Block:** [`2772300`](https://testnet.mintscan.io/noble-testnet/blocks/2772300)
- **Estimated Date & Time:** Wednesday, August 30th, around 11am EST. *(Please note that this is an approximation. Due to varying block times, it's crucial to monitor the chain for the exact upgrade time.)*

## Step-by-Step Upgrade Instructions

> 1. **Set Halt Height:** All validators must set the halt height to `2682700`. There is no on chain instruction to halt.

Reviewer: I don't think we need to set the halt height. 
We have an upgrade handler/name `argon2`

2. **Use Pre-Built Binaries:** Choose the appropriate binary for your system:
   - [AMD](./nobled_linux-amd64)
   - [ARM](./nobled_linux-arm64)

### Using Docker?

You can grab the binaries directly from the latest image with:

```shell

> docker pull ghcr.io/strangelove-ventures/heighliner/noble:v4.0.0-alpha3

Reviewer: Just a note, but these images are pushed to both `heighliner` and the noble repo. So we can have them pull from either one. 
Maybe `noble` is more intuitive? 
https://github.com/strangelove-ventures/noble/pkgs/container/noble/123165979?tag=v4.0.0-alpha3

# Upgrade to v4.0.0-alpha3

## Overview

We are excited to announce the next software upgrade for the Noble `grand-1` testnet chain. This upgrade will transition the network to `v4.0.0-alpha3`. This upgrade adds TBD.

**Key Details:**

> - **Target Block:** [`2682700`](https://testnet.mintscan.io/noble-testnet/blocks/2682700)

Reviewer 2: ```suggestion
- **Target Block:** [`2772300`](https://testnet.mintscan.io/noble-testnet/blocks/2772300)
```

> - **Estimated Date & Time:** Thursday, August 24th, around 11am EST. *(Please note that this is an approximation. Due to varying block times, it's crucial to monitor the chain for the exact upgrade time.)*

Reviewer 2: ```suggestion
- **Estimated Date & Time:** Wednesday, August 30th, around 11am EST. *(Please note that this is an approximation. Due to varying block times, it's crucial to monitor the chain for the exact upgrade time.)*
```

## Step-by-Step Upgrade Instructions

1. **Set Halt Height:** All validators must set the halt height to `2682700`. There is no on chain instruction to halt.
2. **Use Pre-Built Binaries:** Choose the appropriate binary for your system:
   - [AMD](./nobled_linux-amd64)
   - [ARM](./nobled_linux-arm64)

> ## Update
> On 8/25/23 we identified that the AMD binary was wrong. We have updated the binary and the checksums below. To resolve this issue we would like to do another halt-height upgrade at block [2700000](https://testnet.mintscan.io/noble-testnet/blocks/2700000)

Reviewer 2: ```suggestion
```
